### ISSUE TITLE
Support FIO sequential write test case with Alluxio Fuse

### ISSUE THREAD

**Issue by yyongycy** [Alluxio/alluxio#15569]
**Is your feature request related to a problem? Please describe.**
So far Alluxio Fuse doesn't work well with FIO. This is due to some special IO operations, and Alluxio Fuse doesn't support that well. This feature is important as people want to do an apple-to-apple comparison. FIO is a popular tool in posix.

**Describe the solution you'd like**
There are two options:
1. try to cater the create/open operations from FIO in Alluxio Fuse
2. modify one existing IO engine to make the create/open friendly for Alluxio Fuse

**Describe alternatives you've considered**
Like mentioned above item2, copy and create a new IO Engine to adapter to Alluxio Fuse style.

**Urgency**
This is just for a benchmark test, it is not that urgent.

**Additional context**
Some Reference Information:
**FIO parameters**
[root@ip-172-31-8-187 fio_pattern]# cat ~/test_script/fio/fio.cfg
[global]
direct=1
fallocate=none
#ioengine=libaio
group_reporting
time_based
size=5g
#nrfiles=50
iodepth=32
stonewall
runtime=600

[bs2m_job32_write]
directory=/mnt/dummy
bs=2m
numjobs=1
rw=write

**FIO system call**
open("/mnt/dummy/bs2m_job32_write.0.0", **O_WRONLY|O_CREAT**, 0644) = 6
open("/mnt/dummy/bs2m_job32_write.0.0", **O_RDWR|O_CREAT|O_DIRECT**, 0600) = 6

**FUSE received call**
create flags: 0x8041 /bs2m_job32_write.0.0 0100644 umask=0022
open flags: 0xc002 /bs2m_job32_write.0.0

**Comment by LuQQiu** [comment]
@yyongycy The scope of this issue may be big
I tested mainly two workloads

* single thread sequential read a 4GB file with buffer size 4M
fio --name=sequential-read --directory=/mnt/alluxio-fuse --rw=read --refill_buffers --bs=4M --size=4G
fio: pid=0, err=95/file:filesetup.c:182, func=ftruncate, error=Operation not supported

```
"Create an empty file, use truncate() to set the file size to a large value

Fuse.Getattr(path=/sequential-read.0.0)
Fuse.Create(path=/sequential-read.0.0,mode=100644) create flags: 0x8241
Fuse.Truncate(path=/sequential-read.0.0,size=4294967296)
Failed to truncate file /sequential-read.0.0(0 bytes) to 4294967296 bytes: not supported
"
```
This one is hard to support since we need to support using truncate() to set the file size to a larger value which i think is a part of the random write process.

* single thread sequential write a 4GB file with buffer size 4M
fio --name=sequential-write --directory=/mnt/alluxio-fuse --rw=write --refill_buffers --bs=4M --size=4G --end_fsync=1
fio: io_u error on file /mnt/alluxio-fuse/sequential-write.0.0: File exists: write offset=0, buflen=4194304

```
"Create empty file and then open for write
Fuse.Create(path=/sequential-write.0.0,mode=100644)
Fuse.Flush(path=/sequential-write.0.0,fd=1)
Fuse.Release(path=/sequential-write.0.0,fd=1)
Fuse.Open(path=/sequential-write.0.0,flags=0x8002(READ_WRITE))
Fuse.Write(path=/sequential-write.0.0,fd=2,size=131056,offset=0)
Cannot overwrite existing file /sequential-write.0.0 without O_TRUNC flag or fuse.truncate(size=0), flag 0x8002"
```
This one can be solved by my PR #15740

**Comment by LuQQiu** [comment]
* single thread sequential write a 4GB file with buffer size 4M
fio --name=sequential-write --directory=/mnt/alluxio-fuse --rw=write --refill_buffers --bs=4M --size=4G --end_fsync=1
fio: io_u error on file /mnt/alluxio-fuse/sequential-write.0.0: File exists: write offset=0, buflen=4194304

```
"Create empty file and then open for write
Fuse.Create(path=/sequential-write.0.0,mode=100644)
Fuse.Flush(path=/sequential-write.0.0,fd=1)
Fuse.Release(path=/sequential-write.0.0,fd=1)
Fuse.Open(path=/sequential-write.0.0,flags=0x8002(READ_WRITE))
Fuse.Write(path=/sequential-write.0.0,fd=2,size=131056,offset=0)
Cannot overwrite existing file /sequential-write.0.0 without O_TRUNC flag or fuse.truncate(size=0), flag 0x8002"
```
This one can be solved by my PR #15740

**Comment by LuQQiu** [comment]
Tested https://github.com/Alluxio/alluxio/pull/15742 with the following test cases, they can all passed
```
#16 threads sequential read 16 4GB files with buffer size 4M
fio --name=big-file-multi-read --directory=/mnt/alluxio-fuse --rw=read --refill_buffers --bs=4M --size=4G --numjobs=16
#16 threads sequential write 16 4GB files with buffer size 4M
fio --name=big-file-multi-write --directory=/s3fs --rw=write --refill_buffers --bs=4M --size=4G --numjobs=16 --end_fsync=1
```
